virttest/env_process.py

```diff
@@ -1072,10 +1072,13 @@ def preprocess(test, params, env):
 
     # change the kernel params if configured
     if kernel_extra_params_add or kernel_extra_params_remove:
-        for vm in env.get_all_vms():
-            if vm:
-                if not vm.is_alive():
-                    vm.start()
+        if params.get("start_vm", "no") == "yes":
```

Comment:
Am not sure when will we endup in this situation, as we do start the vm here `process(test, params, env, preprocess_image, preprocess_vm)`  if right params are set in config `create_vm_libvirt` for libvirt and `start_vm` for both qemu/libvirt ?

```
  # Preprocess all VMs and images
    if params.get("not_preprocess", "no") == "no":
        process(test, params, env, preprocess_image, preprocess_vm)

    # change the kernel params if configured
    if kernel_extra_params_add or kernel_extra_params_remove:
        if params.get("start_vm", "no") == "yes":
            for vm in env.get_all_vms():
                if vm and not vm.is_alive():
                    if params.get("vm_type") == "libvirt":
                        vm.start()
                    elif params.get("vm_type") == "qemu":
                        vm.create(params=params)
                utils_test.update_boot_option(vm, args_added=kernel_extra_params_add,
                                              args_removed=kernel_extra_params_remove)```

Reply:
probably we can set start_vm and other relavant params if we see `kernel_extra_params_add` and `kernel_extra_params_remove` are set just before `process(test, params, env, preprocess_image, preprocess_vm)` and let framework do the magic?

Reply:
I have taken care of it. Thanks for the suggesion :+1: